### PR TITLE
SEAB-6000: Track GitHub App UNINSTALL events

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
The `installation_repositories` GitHub event can describe either additions or removals, depending on the value of its "action" property.  This PR modifies `handleGitHubInstallation` to handle the "removed" action of a propagated GitHub `installation_repositories` event and adds a corresponding `UNINSTALL` lambda event type.

The corresponding lambda PR is https://github.com/dockstore/lambda/pull/161

**Review Instructions**
Add the qa GitHub App to a repo, wait five minutes, and confirm that the corresponding INSTALL entry appears in the apps logs.  Then, remove the GitHub App from the repo, wait five minutes, and confirm that the corresponding UNINSTALL entry appears in the apps logs.   Confirm that in the apps logs, "Uninstall" is properly capitalized.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-6000

**Security and Privacy**

No concerns.

- [x] Security and Privacy assessed

e.g. Does this change...
* Any user data we collect, or data location?
* Access control, authentication or authorization?
* Encryption features?

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/core/webhook/InstallationRepositoriesPayload.java
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/core/webhook/InstallationRepositoriesPayload.java:60]: TIL they're working on an openapi 4 for things like this (i.e. either this parameter or this other one https://github.com/OAI/OpenAPI-Specification/issues/256#issuecomment-1908868607 )

i.e. it's a little weird that both `repositoriesRemoved` and `repositoriesAdded` are required

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java:2145]: Share an enum with `dockstore-integration-testing/src/test/java/io/dockstore/webservice/helpers/GitHubAppHelper.java` so that if one is changed, the future me knows to change the other at the same time

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java
Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java:2187]: "Hide whitespace" option works well too

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java
Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java:2145]: And link to `installation_repositories` I suppose

**Review by coverbeck — COMMENTED** [review]
Looks good, but I like Denis' suggestion of an enum, so I'll wait for that change for final approval.

**Review by kathy-t — COMMENTED** [review]
Was here, I agree with Denis' feedback so will wait for that

**Review by coverbeck — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/helpers/GitHubAppHelper.java
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/helpers/GitHubAppHelper.java:59]: Could import `io.dockstore.webservice.core.webhook.InstallationRepositoriesPayload` to make this a bit more concise, not worth changing unless there are other changes

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**
